Application/Features/Commands/DeleteMovie/DeleteMovieCommandHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Application.Exceptions;
+using Application.Interfaces;
+using MediatR;
+
+namespace Application.Features.Commands.DeleteMovie;
+
+public sealed class DeleteMovieCommandHandler : IRequestHandler<DeleteMovieCommand>
+{
+    private readonly IDataContext _context;
+
+    public DeleteMovieCommandHandler(IDataContext context)
+    {
+        this._context = context;
+    }
+
+    public async Task Handle(DeleteMovieCommand request, CancellationToken cancellationToken)
+    {
+        var movie = await _context.Movies.FindAsync(request.Id);
```

Comment:
What is the appropriate way to search for a movie. What if we have more than 1 movie with same id